be1-go/hub/socket.go

```diff
@@ -33,32 +36,32 @@ type Client struct {
 	Wait sync.WaitGroup
 }
 
-// NewClient returns an instance of a Client.
-func NewClient(h Hub, conn *websocket.Conn) *Client {
-	return &Client{
+// NewSocket returns an instance of a Socket.
+func NewSocket(h Hub, conn *websocket.Conn) *Socket {
+	return &Socket{
 		hub:  h,
 		conn: conn,
 		send: make(chan []byte, 256),
 		Wait: sync.WaitGroup{},
 	}
 }
```

Comment:
`NewSocket` doesn't initialize `socketType`, What is the type of the created socket? Did you intend that `NewSocket`creates a "default" socket that is none of {client,organizer,witness}?
Also, you create some socket in the file `hub/socket_types.go` (`NewClientSocket` etc), is the method `NewSocket`here necessary at all?

Reply:
Indeed it is not necessary. When refactoring I simply changed the client type and all of its methods to match the socket type and did not think to remove this function. I will make sure to be more rigorous in the future.

Reply:
Awesome, thanks!